x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -546,3 +552,18 @@ func TestRandomizedJoinPoolExitPoolInvariants(t *testing.T) {
 		testPoolInvariants()
 	}
 }
+
+func assertExpectedSharesErrRatio(t *testing.T, expectedShares, actualShares sdk.Int) {
```

Comment:
We should also ensure that `actualShares <= expectedShares` right?

Reply:
How serious is this? I think some tests were failing when I was trying to ensure this but would need to double-check. Let me keep this an issue and investigate it separately

Reply:
https://github.com/osmosis-labs/osmosis/issues/1733